internal/report/chart.go

```diff
@@ -146,5 +152,15 @@ func generateChartData(s *db.Statistics) (
 		}
 	}
 
+	// Fix label for GraphQL if it is unavailable
+	if !s.IsGraphQLAvailable && containsApiCat {
+		for i := 0; i < len(apiIndicators); i++ {
+			if strings.HasPrefix(apiIndicators[i], "graphql") {
+				apiIndicators[i] = "graphql (unavailable)"
+				apiItems[i] = float32(0)
+			}
+		}
+	}
+
```

Comment:
This block:
```
// Fix label for GraphQL if it is unavailable
...
```
And the block above:
```
// Fix label for gRPC if it is unavailable
...
```
Looks almost identical. Maybe it is a good idea to think about how this logic can be generalized into the function. As for now, it is ok (only 2 occurences), but in the future, if we want to use this primitive 3, 4, 5 ... times, it is better to think about function representation.

Reply:
Fixed - https://github.com/wallarm/gotestwaf/pull/167/commits/60bf140e662611fb65a6840d0fba1092805756f5